docker-app/qfieldcloud/authentication/sso/provider_styles.py

```diff
@@ -0,0 +1,25 @@
+from copy import deepcopy
+
+from django.conf import settings
+from django.http import HttpRequest
+from django.templatetags.static import static
+
+
+class SSOProviderStyles:
+    """Helper class to get the style settings for a provider."""
+
+    def __init__(self, request: HttpRequest):
+        self.request = request
+
+    def get(self, provider_id: str) -> dict:
+        """Get the style settings for a (sub)provider."""
+        style_settings = settings.QFIELDCLOUD_SSO_PROVIDER_STYLES.get(provider_id, {})
```

Comment:
Not sure if I would prefer to just throw in this cases, so people would know what we support and what we dont?

Reply:
My idea here was to make it easy for clients to have robust parsing of these styles.
There will always be a `style` property, but possibly an empty dict. Anything below may or may not be there, it's style suggestions if we do have and want to give them. But I don't think a missing style definition for a newly configured provider should prevent login from working. We want as little as possible effort to support new providers, no? Ideally, we (or someone else) can just configure a new provider in the standard django-allauth way, and it will just work.

Reply:
Depends how the client is going to implement it. If they can show sensible button and text, I am ok to leave this optional. But if we already put an expectation that this setting is mandatory, people will suffer a bit more when adding a new provider and a bit less on the client side and future qfc migrations.

I don't have the whole thing in my head right now, so might be missing something.